[BUG] Not compatible with Spring Boot 4.0

**Describe the bug**
Spring boot starters provided by langchain4j are not compatible with SpringBoot 4.0.

**Log and Stack trace**
> Caused by: java.lang.IllegalStateException: Failed to generate bean name for imported class 'dev.langchain4j.openai.spring.AutoConfig'
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.registerBeanDefinitionForImportedConfigurationClass(ConfigurationClassBeanDefinitionReader.java:172)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitionsForConfigurationClass(ConfigurationClassBeanDefinitionReader.java:145)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.loadBeanDefinitions(ConfigurationClassBeanDefinitionReader.java:124)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.processConfigBeanDefinitions(ConfigurationClassPostProcessor.java:458)
	at org.springframework.context.annotation.ConfigurationClassPostProcessor.postProcessBeanDefinitionRegistry(ConfigurationClassPostProcessor.java:310)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanDefinitionRegistryPostProcessors(PostProcessorRegistrationDelegate.java:349)
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:118)
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:794)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:602)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:756)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:445)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:321)
	at org.springframework.boot.test.context.SpringBootContextLoader.lambda$loadContext$2(SpringBootContextLoader.java:156)
	at org.springframework.util.function.ThrowingSupplier.get(ThrowingSupplier.java:58)
	at org.springframework.util.function.ThrowingSupplier.get(ThrowingSupplier.java:46)
	at org.springframework.boot.SpringApplication.withHook(SpringApplication.java:1465)
	at org.springframework.boot.test.context.SpringBootContextLoader$ContextLoaderHook.run(SpringBootContextLoader.java:605)
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:156)
	at org.springframework.boot.test.context.SpringBootContextLoader.loadContext(SpringBootContextLoader.java:115)
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.loadContextInternal(DefaultCacheAwareContextLoaderDelegate.java:247)
	at org.springframework.test.context.cache.DefaultCacheAwareContextLoaderDelegate.lambda$loadContext$0(DefaultCacheAwareContextLoaderDelegate.java:167)
	... 23 more
Caused by: java.lang.IllegalArgumentException: Could not find class [org.springframework.boot.autoconfigure.web.client.RestClientAutoConfiguration]
	at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:353)
	at org.springframework.core.annotation.TypeMappedAnnotation.adapt(TypeMappedAnnotation.java:451)
	at org.springframework.core.annotation.TypeMappedAnnotation.getValue(TypeMappedAnnotation.java:384)
	at org.springframework.core.annotation.TypeMappedAnnotation.asMap(TypeMappedAnnotation.java:273)
	at org.springframework.core.annotation.AbstractMergedAnnotation.asAnnotationAttributes(AbstractMergedAnnotation.java:191)
	at org.springframework.context.annotation.AnnotationBeanNameGenerator.determineBeanNameFromAnnotation(AnnotationBeanNameGenerator.java:143)
	at org.springframework.context.annotation.AnnotationBeanNameGenerator.generateBeanName(AnnotationBeanNameGenerator.java:110)
	at org.springframework.context.annotation.ConfigurationClassBeanDefinitionReader.registerBeanDefinitionForImportedConfigurationClass(ConfigurationClassBeanDefinitionReader.java:168)
	... 43 more
Caused by: java.lang.ClassNotFoundException: org.springframework.boot.autoconfigure.web.client.RestClientAutoConfiguration
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:580)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:490)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:547)
	at org.springframework.util.ClassUtils.forName(ClassUtils.java:302)
	at org.springframework.util.ClassUtils.resolveClassName(ClassUtils.java:343)

**To Reproduce**
Add `langchain4j-spring-boot-starter` dependency to a project using Spring Boot 4.0

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

**Please complete the following information:**
- LangChain4j version:
- LLM(s) used: 
- Java version: 
- Spring Boot version (if applicable): 4.0

**Additional context**

`import org.springframework.boot.autoconfigure.web.client.RestClientAutoConfiguration;`
should be changed to 
`import org.springframework.boot.restclient.autoconfigure.RestClientAutoConfiguration;`
in `dev.langchain4j.openai.spring.AutoConfig`

--- a/docs/docs/tutorials/spring-boot-integration.md
+++ b/docs/docs/tutorials/spring-boot-integration.md
@@ -20,13 +20,26 @@ and other core LangChain4j components through properties.
 To use one of the [Spring Boot starters](https://github.com/langchain4j/langchain4j-spring),
 import the corresponding dependency.
 
-The naming convention for the Spring Boot starter dependency is: `langchain4j-{integration-name}-spring-boot-starter`.
+The naming convention for the Spring Boot starter dependency is:
+- `langchain4j-{integration-name}-spring-boot-starter` for **Spring Boot 3**
+- `langchain4j-{integration-name}-spring-boot4-starter` for **Spring Boot 4**
 
-For example, for OpenAI (`langchain4j-open-ai`), the dependency name would be `langchain4j-open-ai-spring-boot-starter`:
+For example, for OpenAI (`langchain4j-open-ai`):
+
+**Spring Boot 3:**
+ ```xml
+ <dependency>
+     <groupId>dev.langchain4j</groupId>
+     <artifactId>langchain4j-open-ai-spring-boot-starter</artifactId>
+     <version>1.12.2-beta22</version>
+ </dependency>
+```
+
+**Spring Boot 4:**
 ```xml
 <dependency>
     <groupId>dev.langchain4j</groupId>
-    <artifactId>langchain4j-open-ai-spring-boot-starter</artifactId>
+    <artifactId>langchain4j-open-ai-spring-boot4-starter</artifactId>
     <version>1.12.2-beta22</version>
 </dependency>
 ```
@@ -73,11 +86,22 @@ LangChain4j provides a Spring Boot starter for auto-configuring
 [AI Services](/tutorials/ai-services), [RAG](/tutorials/rag), [Tools](/tutorials/tools) etc.
 
 Assuming you have already imported one of the integrations starters (see above),
-import `langchain4j-spring-boot-starter`:
+import `langchain4j-spring-boot-starter` (Spring Boot 3) or `langchain4j-spring-boot4-starter` (Spring Boot 4):
+
+**Spring Boot 3:**
 ```xml
 <dependency>
     <groupId>dev.langchain4j</groupId>
     <artifactId>langchain4j-spring-boot-starter</artifactId>
+    <version>1.12.2-beta22</version>
+</dependency>
+```
+
+**Spring Boot 4:**
+```xml
+<dependency>
+    <groupId>dev.langchain4j</groupId>
+    <artifactId>langchain4j-spring-boot4-starter</artifactId>
     <version>1.12.2-beta22</version>
 </dependency>
 ```
@@ -187,7 +211,8 @@ interface OllamaAssistant {
 In this case, you must explicitly specify **all** components.
 :::
 
-More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java).
+More details can be found [here](https://github.com/langchain4j/langchain4j-spring/blob/main/langchain4j-spring-boot-starter/src/main/java/dev/langchain4j/service/spring/AiService.java)
+(same API for the Spring Boot 4 variant).
 
 ### Listening for AI Service Registration Events
 
@@ -402,9 +427,11 @@ For more details about the `langchain4j-observation` library, please check the [
 
 ## Supported versions
 
-LangChain4j Spring Boot integration requires Java 17 and Spring Boot 3.5, in line with the [Spring Boot OSS support policy](https://spring.io/projects/spring-boot#support).
+LangChain4j Spring Boot integration requires Java 17 and supports both:
+- **Spring Boot 3** (3.5+) — use starters with the `-spring-boot-starter` suffix, in line with the [Spring Boot OSS support policy](https://spring.io/projects/spring-boot#support)
+- **Spring Boot 4** (4.0+) — use starters with the `-spring-boot4-starter` suffix
 
-Support for Spring Boot 4.x is not available yet in LangChain4j, but it's planned for a future release.
+Both families are released together and share the same version number. Choose the set of starters that matches the Spring Boot version in your project.
 
 ## Examples
 - [Low-level Spring Boot example](https://github.com/langchain4j/langchain4j-examples/blob/main/spring-boot-example/src/main/java/dev/langchain4j/example/lowlevel/ChatModelController.java) using [ChatModel API](/tutorials/chat-and-language-models)

